autograder-core/src/main/java/de/firemage/autograder/core/check/naming/ConstantsHaveDescriptiveNamesCheck.java

```diff
@@ -46,6 +47,13 @@ private static boolean isNonDescriptiveStringName(String name, String value) {
             return true;
         }
 
+        // the below code will blow up in size based on the value length
```

Comment:
The code appears overly clever to me. If I understood it correctly, it iterates over all characters in the value and then appends their "string representation" with and without an underscore to all possible existing options. This effectively builds all in-order combinations from the individual options separated by epsilon or an underscore.
This algorithm is *exponential*. On my computer it takes 14 seconds for a string with 11 elements and fails to compute 12 in a reasonable timeframe.

If you ignore multiple and trailing/leading underscores, this can be expressed in a much quicker way. This starts taking 14 seconds for ~2800 chars, which is a lot more practical than 11. If you actually want to *optimize* this, the current values should lend itself to a very simple *linear* DFA instead, incrementally matching the input instead of building an exponential set of possible combinations and then trying to find the input in it. The DFA breaks down when the char options are not prefix-free, but they currently are.

Quicker exp. solution under the above constraints:
```java
  private static boolean isNonDescriptiveStringName(String name, String value) {
    if (NON_DESCRIPTIVE_NAMES.contains(name.toLowerCase())) {
      return true;
    }

    Stream<String> options = Stream.of("");
    if (value.isEmpty()) {
      options = Stream.of("empty", "blank");
    } else {
      for (char c : value.toCharArray()) {
        var charOptions = listCharOptions(c);
        if (charOptions == null) {
          return false;
        }
        options = options.flatMap(
            suffix -> charOptions.stream().map(o -> suffix + o)
        );
      }
    }
    String cleanedName = name.toLowerCase(Locale.ROOT).replace("_", "");
    return options.anyMatch(option -> option.equals(cleanedName));
  }
```